bundle/phases/deploy.go

```diff
@@ -215,5 +219,92 @@ func Deploy(ctx context.Context, b *bundle.Bundle, outputHandler sync.OutputHand
 		return diags
 	}
 
+	logTelemetry(ctx, b)
 	return diags.Extend(bundle.Apply(ctx, b, scripts.Execute(config.ScriptPostDeploy)))
 }
+
+func logTelemetry(ctx context.Context, b *bundle.Bundle) {
+	resourcesCount := int64(0)
+	_, err := dyn.MapByPattern(b.Config.Value(), dyn.NewPattern(dyn.Key("resources"), dyn.AnyKey(), dyn.AnyKey()), func(p dyn.Path, v dyn.Value) (dyn.Value, error) {
+		resourcesCount++
+		return v, nil
+	})
+	if err != nil {
+		log.Debugf(ctx, "failed to count resources: %s", err)
+	}
+
+	jobsIds := make([]string, 0)
+	for _, job := range b.Config.Resources.Jobs {
+		// Do not include missing IDs in telemetry. We can still detect them
+		// by comparing against the resource count.
+		if job.ID == "" {
+			continue
+		}
+		jobsIds = append(jobsIds, job.ID)
+	}
+	pipelineIds := make([]string, 0)
+	for _, pipeline := range b.Config.Resources.Pipelines {
+		// Do not include missing IDs in telemetry. We can still detect them
+		// by comparing against the resource count.
+		if pipeline.ID == "" {
+			continue
+		}
+		pipelineIds = append(pipelineIds, pipeline.ID)
+	}
+	clusterIds := make([]string, 0)
+	for _, cluster := range b.Config.Resources.Clusters {
+		// Do not include missing IDs in telemetry. We can still detect them
+		// by comparing against the resource count.
+		if cluster.ID == "" {
+			continue
+		}
+		clusterIds = append(clusterIds, cluster.ID)
+	}
+	dashboardIds := make([]string, 0)
+	for _, dashboard := range b.Config.Resources.Dashboards {
+		// Do not include missing IDs in telemetry. We can still detect them
+		// by comparing against the resource count.
+		if dashboard.ID == "" {
+			continue
+		}
+		dashboardIds = append(dashboardIds, dashboard.ID)
+	}
+
+	// sort the IDs to make the record generated deterministic
+	// this is important for testing purposes
+	slices.Sort(jobsIds)
+	slices.Sort(pipelineIds)
+	slices.Sort(clusterIds)
+	slices.Sort(dashboardIds)
+
+	// If the bundle UUID is not set, we use a default 0 value.
+	bundleUuid := "00000000-0000-0000-0000-000000000000"
```

Comment:
We could also consider generating a random uuid and hard-code it there. Then we know where it comes from. Zero uuid can be injected elsewhere.

Reply:
>  Even if 0 is injected from some intermediate system that's fine since it still represents the nil value.

but in that case you cannot distinguish intermediate 0 and user-supplied 0.

Reply:
> but in that case you cannot distinguish intermediate 0 and user-supplied 0.

Why would users ever set the bundle UUID to `00000000-0000-0000-0000-000000000000`? It's fine if we lose data for that scenario. Having a clean representation for this value being unset is worth not accounting for that case.